packages/playground/website/src/github/preview-pr/form.tsx

```diff
@@ -0,0 +1,233 @@
+import React, { useEffect } from 'react';
+import { useState } from 'react';
+import { Spinner, TextControl } from '@wordpress/components';
+import css from './style.module.css';
+import { logger } from '@php-wasm/logger';
+import { setActiveSiteError } from '../../lib/state/redux/slice-ui';
+import { useDispatch } from 'react-redux';
+import { PlaygroundDispatch } from '../../lib/state/redux/store';
+import ModalButtons from '../../components/modal/modal-buttons';
+
+interface PreviewPRFormProps {
+	onImported: () => void;
+	onClose: () => void;
+	target: 'wordpress' | 'gutenberg';
+}
+
+const urlParams = new URLSearchParams(window.location.search);
+
+export const targetParams = {
+	'wordpress': {
+		repo: 'wordpress-develop',
+		workflow: 'Test%20Build%20Processes',
+		artifact: 'wordpress-build-'
+	},
+	'gutenberg': {
+		repo: 'gutenberg',
+		workflow: 'Build%20Gutenberg%20Plugin%20Zip',
+		artifact: 'gutenberg-plugin'
+	}
+}
+
+export default function PreviewPRForm({
+	onImported,
+	onClose,
+	target = 'wordpress'
+}: PreviewPRFormProps) {
+	const dispatch: PlaygroundDispatch = useDispatch();
+	const [value, setValue] = useState<string>('');
+	const [submitting, setSubmitting] = useState<boolean>(false);
+	const [errorMsg, setError] = useState<string>('');
+
+	useEffect(() => {
+		const query = new URLSearchParams(window.location.search);
+		if (query.has('core-pr')) {
+			const prNumber = query.get('core-pr');
+			prNumber && setValue(prNumber);
+		}
+	}, []);
+
+	async function handleSubmit(e: React.MouseEvent<HTMLButtonElement>) {
+		e.preventDefault();
+
+		if (!value) {
+			return;
+		}
+
+		await previewPr(value);
+	}
+
+	function renderRetryIn(retryIn: number) {
+		setError(`Waiting for GitHub to finish building PR ${value}. This might take 15 minutes or more! Retrying in ${
+			retryIn / 1000
+		}...`);
+	}
+
+	async function previewPr(prValue: string) {
+		let cleanupRetry = () => {};
+		if (cleanupRetry) {
+			cleanupRetry();
+		}
+
+		let prNumber: string = prValue;
+		setSubmitting(true);
+
+		// Extract number from a GitHub URL
+		if (
+			prNumber
+				.toLowerCase()
+				.includes('github.com/wordpress/wordpress-develop/pull') ||
+			prNumber
+				.toLowerCase()
+				.includes('github.com/wordpress/gutenberg/pull')
+		) {
+			// @ts-ignore
+			prNumber = prNumber.match(/\/pull\/(\d+)/)[1];
+		}
+
+		// Verify that the PR exists and that GitHub CI finished building it
+		// @ts-ignore
+		const zipArtifactUrl = `https://playground.wordpress.net/plugin-proxy.php?org=WordPress&repo=${targetParams[target].repo}&workflow=${targetParams[target].workflow}&artifact=${targetParams[target].artifact}${target === 'wordpress' ? prNumber : ''}&pr=${prNumber}`;
```

Comment:
I think this is code derived from an existing file [like `wordpress.html`](https://github.com/ajotka/wordpress-playground/blob/4a74fe8bdd3f657ab686f869dc0d7b8ed4f47731/packages/playground/website/public/wordpress.html#L159).

But I agree that using `append()` would help readability and is also good practice to avoid URL encoding issues (which may or may not be a real risk in this context).